Sofa/framework/LinearAlgebra/src/sofa/linearalgebra/RotationMatrix.cpp

```diff
@@ -203,47 +203,52 @@ void RotationMatrix<Real>::opMulTM(linearalgebra::BaseMatrix * bresult,linearalg
     linearalgebra::BaseMatrix::opMulTM(bresult,bm);
 }
 
+template<class Real>
+template<typename real2>
+void RotationMatrix<Real>::rotateSparseMatrix(
+    linearalgebra::BaseMatrix * result,
+    const SparseMatrix<real2>* Jmat)
+{
+    for (const auto& [rowId, row] : *Jmat)
+    {
+        for (auto it = row.begin(); it != row.end();)
+        {
+            const auto colId = it->first;
+
+            const auto v0 = it->second;
+            ++it;
+            if (it == row.end())
```

Comment:
At first sight, I have the impression the inner loop looks weird and it is doing the calculus only if row.size() >= 3. I wonder what happens in other cases.

Reply:
Clearly this matrix has been designed for 3d DOFs.
I don't think the condition `row.size()% 3 != 0` is handled properly. It should probably be checked earlier.

Reply:
I look at the original code, which is probably more clear on the fact that the row iterator is not constant time. This is probably why algorithm is written that way. But with modern c++ this is less obvious.

Reply:
I don't see how it is clearer. Would writing the types instead of `auto` help?

Reply:
It just took me more time to understand this is not iterating over a non standard structure that is exposed through a key-value sub-structure.  Not using auto here emphasize that much more.

Reply:
I think it is now clearer that the iterator comes from a `std::map`. I also use structured binding on the iterator.
Contrary to the initial algorithm, I use directly `{colId0, colId1, colId2}` instead of `{colId+0, colId+1, colId+2}`. The assumption that they were equal in the initial algorithm was not explicit. I added an assert for that.
A comment is also added.
Let me know what you think

Reply:
I changed the code again. Now the out of bounds check is no longer necessary. It simplifies the inner-loop code.